datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java

```diff
@@ -73,16 +73,30 @@ class CoordinatorEventBlockingQueue implements MetricsAware {
 
 
   /**
-   * Add a single event to the queue, overwriting events with the same name and same metadata.
+   * Add a single event to the queue. Defaults to adding the event at the end of the queue.
    * @param event CoordinatorEvent event to add to the queue
    */
   public synchronized void put(CoordinatorEvent event) {
-    LOG.info("Queuing event {} to event queue", event.getType());
+    put(event, true);
```

Comment:
When do we need to support for inserting at rear?

Reply:
All the events to this coordinator queue are inserted in the rear. The only case in which we have to insert in the front is what the PR proposes.